components/pages/workPageLayout/helper.ts

```diff
@@ -101,3 +119,55 @@ export const getIconNameFromMaterialType = (materialType: GeneralMaterialTypeCod
     return "podcast"
   }
 }
+
+export const slideSelectOptionsFromMaterialTypes = (workMaterialTypes: GeneralMaterialType[]) => {
+  return workMaterialTypes.map(materialType => {
+    return {
+      code: materialType.code,
+      display: translateMaterialTypesStringForRender(
+        materialType.code as GeneralMaterialTypeCodeEnum
+      ),
+    }
+  }) as SlideSelectOption[]
+}
+
+export const sortSlideSelectOptions = (options: SlideSelectOption[]) => {
+  return options.sort((a, b) => {
+    // sort by the index of the GeneralMaterialTypeCodeEnum in the materialTypeSortPriority array
+    return (
+      goConfig("materialtypes.sortpriority").indexOf(a.code) -
+      goConfig("materialtypes.sortpriority").indexOf(b.code)
```

Comment:
You could choose to extract the goConfig once before the sort function. But it is not a biggie and there should be no performance issues bt doing what you have done here.